### PR TITLE
Pathlib

### PR DESCRIPTION
**Pull request by exvertus**

### CODE REVIEW

File: content/posts/pathlib.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+from os.path import join
+path = join(basepath, 'a', 'better', 'path')
+{{</ highlight >}}
+
+But in terms of ease of use, these libraries are starting to show their age.
+Python3 includes [pathlib](https://docs.python.org/3/library/pathlib.html), a more convenient class-based library for interacting with paths.
+
+##### Pathlib Classes
+
+By importing pathlib and using a Path class, we'll get a *concrete* class based on the underlying filesystem.
```
Comment (exvertus) [inline content/posts/pathlib.md:34]: @AnUnpronounceableName is this an example of the "pathlib" font inconsistency you were mentioning?

Comment (AnUnpronounceableName) [reply content/posts/pathlib.md:34]: This is one example, yeah

File: content/posts/pathlib.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+In my case, I'm using Windows. Testing in a Python console will return a WindowsPath.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:35]: You could do `WindowsPath` here

File: content/posts/pathlib.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+{{< highlight python >}}
+>>> from pathlib import Path
+>>> pathlib_path = Path.cwd()
+>>> type(pathlib_path)
+<class 'pathlib.WindowsPath'>
+>>> print(pathlib_path)
+C:\Users\Tom\AppData\Local\Programs\Python\Python39
+{{</ highlight >}}
+
+Pathlib also has a PosixPath concrete class that you'll get from calling ```Path()``` on an Ubuntu machine, for instance.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:46]: - `Pathlib` at the beginning of the sentence
- `PosixPath`
- Could potentially do `concrete` class but I'm unsure if you're trying to go full coding style for every possible thing here.

File: content/posts/pathlib.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+>>> from pathlib import Path
+>>> pathlib_path = Path.cwd()
+>>> type(pathlib_path)
+<class 'pathlib.WindowsPath'>
+>>> print(pathlib_path)
+C:\Users\Tom\AppData\Local\Programs\Python\Python39
+{{</ highlight >}}
+
+Pathlib also has a PosixPath concrete class that you'll get from calling ```Path()``` on an Ubuntu machine, for instance. 
+Each concrete class is inherited from a PurePath parent, and each PurePath class allows path operations, provided they don't touch the filesystem, which will error.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:47]: - Could do `PurePath`

File: content/posts/pathlib.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+File "<pyshell#63>", line 1, in <module>
+pure_linux_path.rmdir()
+AttributeError: 'PurePosixPath' object has no attribute 'rmdir'
+{{</ highlight >}}
+
+This can be an elegant way to do some filepath manipulation in the opposite platform; a necessary evil that I've sometimes run into for cross-platform CI projects.
+
+##### Console comparisons with os
+
+One pet-peeve of mine, especially when revisiting Python after some time away, is that os.path contains functions instead of methods.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:65]: - `os.path`

File: content/posts/pathlib.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+AttributeError: 'PurePosixPath' object has no attribute 'rmdir'
+{{</ highlight >}}
+
+This can be an elegant way to do some filepath manipulation in the opposite platform; a necessary evil that I've sometimes run into for cross-platform CI projects.
+
+##### Console comparisons with os
+
+One pet-peeve of mine, especially when revisiting Python after some time away, is that os.path contains functions instead of methods. 
+It is easy to forget that as my little whoops below demonstrates. 
+The OOP consistency from pathlib avoids this.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:67]: - `pathlib`

File: content/posts/pathlib.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+>>> from pathlib import Path
+>>> pathlib_path = Path.cwd()
+>>> pathlib_path.exists()
+True
+{{</ highlight >}}
+{{</ div/column >}}
+{{</ div/row >}}
+
+Another source of errors is the inconsistent interface.
+Directories have to do with paths, so the function for listing them must be in os.path, right?
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:93]: - `os.path`

File: content/posts/pathlib.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+{{< highlight python >}}
+>>> list(pathlib_path.iterdir())
+[WindowsPath('C:/Users/Tom/AppData/Local/Programs/Python/Python39/DLLs'), 
+ WindowsPath('C:/Users/Tom/AppData/Local/Programs/Python/Python39/Doc'),
+ ...]
+{{</ highlight >}}
+{{</ div/column >}}
+{{</ div/row >}}
+
+Pathlib also provides some convenient attributes, retrieving values related to the path is as simple as it should be.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:116]: - `PathLib`

File: content/posts/pathlib.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+Some of the terminology pathlib uses for paths can be quickly understood by looking at the [pathlib cheatsheet](https://github.com/chris1610/pbpython/blob/master/extras/Pathlib-Cheatsheet.pdf).
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:117]: - `pathlib`

File: content/posts/pathlib.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+##### Final examples
+
+So while it is still important to avoid code like this:
+
+{{< highlight python >}}
+path = basepath + '/never' + '/do' + '/this'
+{{</ highlight >}}
+
+It is easy to understand the temptation, which brings me to my conclusion:
+Pathlib enables me to think about paths the way I already do, *without* the hurdles of a dispersed interface.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:152]: - `pathlib`

File: content/posts/pathlib.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+Below I've simplified a scenario I've encountered working on a CI project, again comparing os with the same logic refactored for pathlib.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:153]: - `pathlib`

File: content/posts/pathlib.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+{{< highlight python >}}
+from pathlib import Path
+workspace = Path(os.environ.get('WORKSPACE', '.'))
+(workspace/'build'/'info').mkdir(parents=True, exist_ok=True)
+project.build()
+xml_results = (workspace/'build'/'info'/'results.xml').read_text()
+build_log = (workspace/'build'/'info'/'build_log.txt').read_text()
+{{</ highlight >}}
+
+With os, I'm almost forced to over-name things to keep the verbosity down, hence the 'info_folder' variable.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:175]: - `os`

File: content/posts/pathlib.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+There really isn't a need for such a variable when using pathlib.
```
Comment (AnUnpronounceableName) [inline content/posts/pathlib.md:176]: - `pathlib`

**Pull request merged**